core/src/cluster_info.rs

```diff
@@ -1040,7 +1036,21 @@ impl ClusterInfo {
                     if exit.load(Ordering::Relaxed) {
                         return;
                     }
-                    obj.write().unwrap().purge(timestamp());
+                    let timeout = {
+                        if let Some(ref bank_forks) = bank_forks {
+                            let bank = bank_forks.read().unwrap().working_bank();
+                            let epoch = bank.epoch();
+                            let epoch_schedule = bank.epoch_schedule();
+                            let ms_per_slot = (1.0f64 / DEFAULT_TICKS_PER_SECOND as f64
+                                * DEFAULT_TICKS_PER_SLOT as f64
+                                * 1_000f64) as u64;
+                            epoch_schedule.get_slots_in_epoch(epoch) * ms_per_slot
```

Comment:
Should this be min of default timeout and this new computed value?

Reply:
We want the longer timeout for stakes only.

Reply:
Ah okay I misread this to mean "get remaining slots in epoch" in which case I wanted it to he capped at the bottom to the default timeout. Nvm. Misread it.

Reply:
is there a reason you can think of if that working bank would be NULL randomly?

Reply:
It might be at the very beginning since gossip comes up before the validator's replay stage. It shouldn't keep going back and forth though.

Reply:
yea, seems fine from a test run.